backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/cache/ThreadLocalCacheController.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * #%L
+ * de.metas.adempiere.adempiere.base
+ * %%
+ * Copyright (C) 2022 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.cache;
+
+import com.google.common.collect.ImmutableSet;
+import lombok.NonNull;
+import org.adempiere.util.lang.IAutoCloseable;
+
+public final class ThreadLocalCacheController
+{
+	public static final ThreadLocalCacheController instance = new ThreadLocalCacheController();
+
+	@NonNull
+	private final ThreadLocal<Boolean> noCacheRef = new ThreadLocal<>();
+
+	private ThreadLocalCacheController()
+	{
+	}
+
+	public static boolean computeAllowDisablingCache(final String cacheName, final ImmutableSet<CacheLabel> labels)
+	{
+		return !isApplicationDictionaryCache(cacheName, labels);
+	}
+
+	private static boolean isApplicationDictionaryCache(final String cacheName, final ImmutableSet<CacheLabel> labels)
+	{
+		final boolean anyApplicationDictionaryTableNames = labels.stream().anyMatch(CacheLabel::isApplicationDictionaryTableName);
+		if (!anyApplicationDictionaryTableNames)
+		{
+			return false;
+		}
+
+		final boolean anyNonApplicationDictionaryTableNames = labels.stream().anyMatch(label -> isNonApplicationDictionaryTableName(label, cacheName));
+		return !anyNonApplicationDictionaryTableNames;
+	}
+
+	private static boolean isNonApplicationDictionaryTableName(final CacheLabel label, final String cacheName)
+	{
+		return label.equalsByName(cacheName) //ignore the label created from this.cacheName as it's not necessary a table name
```

Comment:
I think we are missing a `!` -> should be `return !label.equalsByName(cacheName)`